pysmt/test/test_bv_simplification.py

```diff
@@ -94,6 +95,316 @@ def all_special(self):
                 for op in self.special_operators:
                     self.check(op(l, n))
 
+    def test_bv_0_add(self):
+        x = Symbol("x", BVType(32))
+        f = BVAdd(BVZero(32), x)
+        self.assertEqual(f.simplify(), Symbol("x", BVType(32)))
+
+    def test_bv_add_0(self):
+        x = Symbol("x", BVType(32))
+        f = BVAdd(x, BVZero(32))
+        self.assertEqual(f.simplify(), Symbol("x", BVType(32)))
+
+    def test_bv_add_constants(self):
+        f = BVAdd(BV(10, 32), BV(12, 32))
+        self.assertEqual(f.simplify(), BV(22, 32))
+
+    def test_bv_add_symbols(self):
+        x, y = (Symbol(name, BVType(32)) for name in "xy")
+        f = BVAdd(x, y)
+        self.assertEqual(f.simplify(), BVAdd(x, y))
+
+    def test_bv_0_mul(self):
+        x = Symbol("x", BVType(32))
+        f = BVMul(BVZero(32), x)
+        self.assertEqual(f.simplify(), BVZero(32))
+
+    def test_bv_1_mul(self):
+        x = Symbol("x", BVType(32))
+        f = BVMul(BVOne(32), x)
+        self.assertEqual(f.simplify(), Symbol("x", BVType(32)))
+
+    def test_bv_mul_0(self):
+        x = Symbol("x", BVType(32))
+        f = BVMul(x, BVZero(32))
+        self.assertEqual(f.simplify(), BVZero(32))
+
+    def test_bv_mul_1(self):
+        x = Symbol("x", BVType(32))
+        f = BVMul(x, BVOne(32))
+        self.assertEqual(f.simplify(), Symbol("x", BVType(32)))
+
+    def test_bv_mul_constants(self):
+        f = BVMul(BV(10, 32), BV(12, 32))
```

Comment:
Does this consider overflow?

Reply:
No, but I'll add tests for overflow cases in the next PR.